core/domain/exp_domain.py

```diff
@@ -2147,7 +2840,11 @@ def _update_gadget_visibilities_for_renamed_state(
 
     def _update_gadget_visibilities_for_deleted_state(self, state_name):
         """Updates the visible_in_states property for gadget instances
-        visible in the deleted state."""
+        visible in the deleted state.
+
+        Args:
+            gadget_name: str. The gadget name to delete.
```

Comment:
This does not match the method signature -- please take a careful look at what this method is doing.